### DISCUSSION TITLE
Bone directions

### DISCUSSION THREAD

**Discussion by BadDogSkyrim** [BadDogSkyrim/PyNifly#153] [Ideas]
Does anyone care what direction the bones point?

I've been adding a rotation to the bones so that in blender, they point in a direction that's relevant to the skeleton--e.g. the thigh bone points to the knee, the knee to the ankle, etc. But that's not the rotation that's in the nif.

There are a number of places that this rotation has to be applied or removed. The bones themselves are fine but I just discovered that if FO4 attachment points are relative to a bone, they are wrong. Animations have the same problem--they depend on the bone orientation.

If I remove this extra rotation, bones generally run perpendicular to the limb. (Not a rule, just how it generally falls out.) So does anyone care if I leave it this way? Not only does it make my life easier, but I prefer that what you see in blender be as close as possible to what's actually in the nif.

![image](https://github.com/BadDogSkyrim/PyNifly/assets/12353950/4d08e784-018a-4d03-ac70-a811bcac3b42)  
![image](https://github.com/BadDogSkyrim/PyNifly/assets/12353950/913dd807-dd36-4f21-ab98-c8f31e172c0d)

**Comment by Shiagur** [comment]
I think it's better to have accurate sideways bones, especially since the Roll value in Blender's bones also seems to influence a lot of stuff. 
Found that out the hard way when I tried to make an animation rig for power armor without 3d max...
And if you only weight paint, which direction they point in is probably not too important.
